app/src/main/java/com/hedvig/app/feature/offer/usecase/getquote/GetQuoteIdsUseCase.kt

```diff
@@ -17,41 +17,31 @@ class GetQuoteIdsUseCase(
     private val offerRepository: OfferRepository,
 ) {
 
-    data class Error(val message: String?)
-
     @JvmInline
     value class QuoteIds(val ids: List<String>)
 
-    suspend operator fun invoke(quoteCartId: String?): Either<Error, QuoteIds> {
+    suspend operator fun invoke(quoteCartId: String?): Either<ErrorMessage, QuoteIds> {
         return if (featureManager.isFeatureEnabled(Feature.QUOTE_CART) && quoteCartId != null) {
             getIdsFromQuoteCart(quoteCartId)
         } else {
             getIdsFromLastQuoteOfMember()
         }
     }
 
-    private suspend fun getIdsFromQuoteCart(quoteCartId: String): Either<Error, QuoteIds> {
-        return when (val result = offerRepository.offer(nonEmptyListOf(quoteCartId)).first()) {
-            is OfferRepository.OfferResult.Error -> Either.Left(Error(null))
-            is OfferRepository.OfferResult.Success -> {
-                val ids = QuoteIds(result.data.quoteBundle.quotes.map { it.id })
-                Either.Right(ids)
+    private suspend fun getIdsFromQuoteCart(quoteCartId: String): Either<ErrorMessage, QuoteIds> {
+        return offerRepository.offer(nonEmptyListOf(quoteCartId))
+            .first()
+            .map {
+                val quoteIds = it.quoteBundle.quotes.map { it.id }
+                QuoteIds(quoteIds)
             }
-        }
     }
 
-    private suspend fun getIdsFromLastQuoteOfMember(): Either<Error, QuoteIds> {
-        return apolloClient
-            .query(LastQuoteIdQuery())
-            .safeQuery()
-            .toEither()
-            .mapLeft { Error(it.message) }
-            .flatMap {
-                Either.conditionally(
-                    it.lastQuoteOfMember.asCompleteQuote?.id != null,
-                    ifFalse = { Error(null) },
-                    ifTrue = { QuoteIds(listOf(it.lastQuoteOfMember.asCompleteQuote!!.id)) }
-                )
-            }
-    }
+    private suspend fun getIdsFromLastQuoteOfMember(): Either<ErrorMessage, QuoteIds> = apolloClient
+        .query(LastQuoteIdQuery())
+        .safeQuery()
+        .toEither { ErrorMessage(it) }
+        .map {
+            QuoteIds(listOf(it.lastQuoteOfMember.asCompleteQuote!!.id))
```

Comment:
Are we like sure sure that this isn't null? Before we were checking it in the `Either.conditionally` but we're not doing this anymore. This isn't safe is it?